### PR TITLE
feat: Custom replication slot name

### PR DESCRIPTION
**Pull request by RobyBen**
Issue #540
**client.py**
Introduced support for specifying replication_slot_name in the replication logic
Added replication=database to the connection string
Updated application_name

**tap.py**
Validation for replication_slot_name: <=60 characters, and do not start with pg_.
Added replication_slot_name as a configurable parameter in config_jsonschema, with a default value of "tappostgres".
Passed replication_slot_name to the PostgresConnector.
Added configuration examples

**Added test_slot_name.py to the tests folder**

### CODE REVIEW

File: tap_postgres/client.py [hunk 1]
```diff
@@ -460,13 +463,20 @@ def logical_replication_connection(self):
         Uses a direct psycopg2 implementation rather than through sqlalchemy.
         """
         connection_string = (
-            f"dbname={self.config['database']} user={self.config['user']} password="
-            f"{self.config['password']} host={self.config['host']} port="
-            f"{self.config['port']}"
+            f"dbname={self.config['database']} "
+            f"user={self.config['user']} "
+            f"password{self.config['password']} "
```
Comment (edgarrmondragon) [inline tap_postgres/client.py:468]:
```suggestion
            f"password={self.config['password']} "
```

File: tap_postgres/tap.py [hunk 2]
```diff
@@ -50,6 +50,23 @@ def __init__(
...
+            assert (
+                slot_name.isalnum() or "_" in slot_name
+            ), "Replication slot name must contain letters, numbers and underscores"
+            max_slot_name_len = 63
+            assert (
+                len(slot_name) <= max_slot_name_len
+            ), "Replication slot name must be less than 63 characters"
+            assert not slot_name.startswith(
+                "pg_"
+            ), "Replication slot name cannot start with 'pg_'"
```
Comment (edgarrmondragon) [inline tap_postgres/tap.py:68]: I think all of these validations can be added to the `Property` instance

File: tap_postgres/tap.py [hunk 3]
```diff
@@ -98,6 +98,20 @@ def __init__(
         )
 
     config_jsonschema = th.PropertiesList(
+        th.Property(
+            "replication_slot_name",
+            th.StringType,
```
Comment (edgarrmondragon) [inline tap_postgres/tap.py:103]:
```suggestion
            th.StringType(
                pattern="^(?!pg_)[A-Za-z0-9_]{1,63}$"
            ),
```

File: tap_postgres/tap.py [hunk 4]
```diff
@@ -98,6 +98,21 @@ def __init__(
         )
 
     config_jsonschema = th.PropertiesList(
+        th.Property(
+            "replication_slot_name",
+            th.StringType(
+                pattern="^(?!pg_)[A-Za-z0-9_]{1,63}$"
```
Comment (edgarrmondragon) [inline tap_postgres/tap.py:104]:
```suggestion
                pattern="^(?!pg_)[A-Za-z0-9_]{1,63}$",
```

File: tests/test_slot_name.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+from tap_postgres.tap import TapPostgres
+
+
+class TestReplicationSlot(unittest.TestCase):
+    def setUp(self):
+        self.default_config = {
+            "host": "localhost",
+            "port": 5432,
+            "dbname": "test_db",
```
Comment (edgarrmondragon) [inline tests/test_slot_name.py:11]:
```suggestion
            "dbname": "test_db",
            "user": "postgres",
```

File: tests/test_slot_name.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            "user": "postgres",
```
Comment (edgarrmondragon) [inline tests/test_slot_name.py:12]:
```suggestion
            "user": "postgres",
            "password": "s3cr3t",
```

**Review by edgarrmondragon — APPROVED** [review]
Thanks @RobyBen!

**Comment by RobyBen** [comment]
@edgarrmondragon. No, thank you, it's been a rewarding experience

**Pull request merged**